File: Examples/VisualBasic/AsposePdfFacades/Annotations/CreateFileAnnotation.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades
Namespace AsposePDFFacades.Annotations
    Public Class CreateFileAnnotation
        Public Shared Sub Run()
            ' ExStart:CreateFileAnnotation
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Annotations()

            ' Instantiate PdfContentEditor object
            Dim editor As New PdfContentEditor()
            ' Bind input PDF file
            editor.BindPdf(dataDir & Convert.ToString("CreateFileAnnotation.pdf"))
            ' The last argumnet is for transparency of icon
            editor.CreateFileAttachment(New System.Drawing.Rectangle(50, 50, 10, 10), "here", dataDir & Convert.ToString("AddFreeTextAnnotation.pdf"), 1, "Paperclip", 0.005)
            ' Save the updated PDF file
            editor.Save(dataDir & Convert.ToString("PdfWith_Transparent_Annotation_out_.pdf"))
            ' ExEnd:CreateFileAnnotation
        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePdfFacades/Annotations/CreateFileAnnotation.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace CSharp.AsposePDFFacades.Annotations
{
    public class CreateFileAnnotation
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Annotations();

            // instantiate PdfContentEditor object
            PdfContentEditor editor = new PdfContentEditor();
            // bind input PDF file
            editor.BindPdf(dataDir+ "CreateFileAnnotation.pdf");
            // the last argumnet is for transparency of icon
            editor.CreateFileAttachment(new System.Drawing.Rectangle(50, 50, 10, 10), "here", "simple.doc", 1, "Paperclip", 0.005);
            // save the updated PDF file
            editor.Save(dataDir+ "PdfWith_Transparent_Annotation.pdf");
            
        }
    }
}